test/arithmetic_test.rb

```diff
@@ -0,0 +1,133 @@
+require "test_helper"
+
+class Measured::ArithmeticTest < ActiveSupport::TestCase
+  setup do
+    @two = Magic.new(2, :magic_missile)
+    @three = Magic.new(3, :magic_missile)
+    @four = Magic.new(4, :magic_missile)
+  end
+
+  test "#+ should add together same units" do
+    assert_equal Magic.new(5, :magic_missile), @two + @three
+    assert_equal Magic.new(5, :magic_missile), @three + @two
+  end
+
+  test "#+ should add a number to the value" do
+    assert_equal Magic.new(5, :magic_missile), @two + 3
+    assert_equal Magic.new(5, :magic_missile), 2 + @three
+  end
+
+  test "#+ should raise if different unit system" do
+    assert_raises TypeError do
+      OtherFakeSystem.new(1, :other_fake_base) + @two
+    end
+
+    assert_raises TypeError do
+      @two + OtherFakeSystem.new(1, :other_fake_base)
+    end
+  end
+
+  test "#+ should raise if adding something nonsense" do
+    assert_raises TypeError do
+      @two + "thing"
+    end
+
+    assert_raises TypeError do
+      "thing" + @two
+    end
+  end
+
+  test "#- should subtract same units" do
+    assert_equal Magic.new(-1, :magic_missile), @two - @three
+    assert_equal Magic.new(1, :magic_missile), @three - @two
+  end
+
+  test "#- should subtract a number from the value" do
+    assert_equal Magic.new(-1, :magic_missile), @two - 3
+    assert_equal Magic.new(1, :magic_missile), 2 - @three
+  end
+
+  test "#- should raise if different unit system" do
+    assert_raises TypeError do
+      OtherFakeSystem.new(1, :other_fake_base) - @two
```

Comment:
Yes you can.

```
pry(main)> Measured::Weight.new(1, :lb) - Measured::Weight.new(1, :g)
=> #<Measured::Weight: 0.99779537738E0 lb>
```

What this is saying is that you can't subtract an inch from a pound.